core/src/main/java/com/digitalpebble/stormcrawler/filtering/basic/BasicURLNormalizer.java

```diff
@@ -303,11 +303,14 @@ public int compare(NameValuePair p1, NameValuePair p2) {
      * @return corrected url
      */
     private String unmangleQueryString(String urlToFilter) {
-        int firstAmp = urlToFilter.indexOf('&');
+        String[] pathElements = urlToFilter.split("/");
+        int firstAmp = pathElements[pathElements.length - 1].indexOf('&');
```

Comment:
I don't recall the complete normalizer code, so the following might not happen (as other methods guard against it...)

If `pathElements` is empty, we will access index `-1`. While this case might not happen in production (assuming every encountered URL starts with a protocol), we should be robust against everything we encounter in the wild. Can we guard it with a `pathElements.length > 0` check?